feat(tests): Restructured and Expanded Integration Tests for CinderGraph

me-hem opened the pull request:
This PR restructures and expands the CinderGraph integration test suite from a single monolithic test file into a modular, well-organized structure that improves maintainability, clarity, and test coverage (issue #118). The previous integration tests (`tests/integration/CinderGraph/CinderGraphShard.cpp`) were minimal and covered only basic operations. The new structure introduces dedicated files for each graph operation and scenario, allowing for more comprehensive testing.

## New Structure

```
tests/integration/CinderGraph/
├── common/                          # Shared test utilities
│   └── DummyGraphBuilder.hpp       # Graph factory methods
│
├── functional/                      # Per-method integration tests
│   ├── addVertex_test.cpp          # Vertex addition tests
│   ├── removeVertex_test.cpp       # Vertex removal tests
│   ├── addEdge_test.cpp            # Edge addition (weighted/unweighted)
│   ├── removeEdge_test.cpp         # Edge removal tests
│   ├── updateEdge_test.cpp         # Edge weight update tests
│   ├── getEdge_test.cpp            # Edge retrieval tests
│   ├── clearEdges_test.cpp         # Edge clearing tests
│   └── hasVertex_test.cpp          # Vertex existence checks
│
└── scenarios/                       # Multi-step integration workflows
    ├── BasicConnectivity_test.cpp   # Basic graph operations flow
    ├── UpdateAndQueryFlow_test.cpp  # Update and query patterns
    ├── BulkInsertionStress_test.cpp # Stress testing with bulk ops
    └── ClearAndReuseGraph_test.cpp  # Graph clearing and reuse
```

## Changes Made

### 1. Common Utilities (common/)

- `DummyGraphBuilder.hpp` - Provides factory methods for creating test graphs with consistent configurations


### 2. Functional Tests (functional/)

- `addVertex_test.cpp`: Vertex insertion, duplicate handling
- `removeVertex_test.cpp`: Vertex deletion, non-existent vertex handling
- `addEdge_test.cpp`: Edge creation for all type combinations
- `removeEdge_test.cpp`: Edge deletion, weight retrieval on removal
- `updateEdge_test.cpp`: Weight updates, failure cases
- `getEdge_test.cpp`: Edge retrieval, optional weight handling
- `clearEdges_test.cpp`: Edge clearing while preserving vertices
- `hasVertex_test.cpp`: Vertex existence queries

### 3. Scenario Tests (scenarios/)

- `BasicConnectivity_test.cpp`: 
  - Add vertices and edges
  - Remove edges and verify connectivity
  - Remove vertices and check cascade effects
  - Complex graph manipulations

- `UpdateAndQueryFlow_test.cpp`:
  - Add → Update → Query patterns
  - Multiple sequential updates
  - Mixed operations with repeated queries
  - Remove → Add → Update cycles

- `BulkInsertionStress_test.cpp`:
  - Bulk vertex insertion 
  - Bulk edge insertion 
  - Dense graph creation
  - Mixed bulk operations with updates

- `ClearAndReuseGraph_test.cpp`:
  - Clear and verify state
  - Reuse after clearing
  - Stress test with repeated clear cycles
  - Empty graph clearing

me-hem committed:
refatored addEdge tests for CinderGraph

me-hem committed:
Adde addVertex tests for CinderGraph

me-hem committed:
Added tests to validate clearEdges functionality for CinderGraph

me-hem committed:
Added tests to validate getEdge functionality for CinderGraph

me-hem committed:
Added tests to validate hasVertex functionality for CinderGraph

me-hem committed:
Added tests to validate removeEdge functionality for CinderGraph

me-hem committed:
Added tests to validate removeVertex functionality for CinderGraph

me-hem committed:
Added tests to validate updateEdge functionality for CinderGraph

me-hem committed:
Refactored Class name for CinderGraph functional tests

me-hem committed:
Added Basic Connectivity Tests for CinderGraph

me-hem committed:
Added bulk Insertion Stress Tests for CinderGraph

me-hem committed:
Added Graph clearing and reusing tests for Cindergraph

me-hem committed:
Added Query Flow and Graph updation tests for Cindergraph

me-hem committed:
Added DummyGraphBuilder and integrated with CinderGraph tests

me-hem committed:
Integration Tests for CinderGraph restructured and expanded

SharonIV0x86 commented on tests/integration/CinderGraph/scenerios/BasicConnectivity_test.cpp:
```diff
@@ -0,0 +1,119 @@
+#include "../common/DummyGraphBuilder.hpp"
```
The folder name needs to be corrected.

me-hem replied:
Changes pushed.

SharonIV0x86 replied:
@me-hem i believe it should be ``scenarios`` not ``scenerios``

me-hem replied:
Oops I missed that

me-hem replied:
@SharonIV0x86 Changes pushed.

me-hem committed:
Header reference for DummyGraphBuilder updated

me-hem commented:
@SharonIV0x86 I have updated all references to the DummyGraphBuilder header file and modified CMakeLists.txt to build this header file too. You can now consider it for review.

Commit:
fixed typo in folder name

SharonIV0x86 reviewed (approved):
Looks good to me.

SharonIV0x86 commented:
@me-hem Thank you for contributing your PR has been merged.
